src/ape/api/providers.py

```diff
@@ -661,8 +661,17 @@ def estimate_gas_cost(self, txn: TransactionAPI, **kwargs) -> int:
 
         Returns:
             int: The estimated cost of gas to execute the transaction
-            reported in the fee-currency's smallest unit, e.g. Wei.
+            reported in the fee-currency's smallest unit, e.g. Wei. If the
+            provider's network has been configured with a gas limit override, it
+            will be returned. If the gas limit configuration is "max" this will
+            return the block maximum gas limit.
         """
+        if isinstance(self.network.gas_limit, int):
+            return self.network.gas_limit
+
+        if self.network.gas_limit == "max":
```

Comment:
I think I found a testing gap, this `if` block here.

Could we add a test in `test_provider` like `test_estimate_gas_cost_when_gas_limit_max_uses_value_from_block`

Reply:
Updated in 002524a3